File: src/main/resources/com/haulmont/cuba/cli/cubaplugin/templates/project/7.2/modules/web/test/$[project.rootPackage]/${project.testContainerPrefix}WebTestContainer.java
package ${project.rootPackage};

import com.haulmont.cuba.web.testsupport.TestContainer;

import java.util.ArrayList;
import java.util.Arrays;

public class ${project.testContainerPrefix}WebTestContainer extends TestContainer {

    public ${project.testContainerPrefix}WebTestContainer() {
        appComponents = new ArrayList<>(Arrays.asList(
                "com.haulmont.cuba"
                // add CUBA add-ons and custom app components here
        ));
        appPropertiesFiles = Arrays.asList(
                // List the files defined in your web.xml
                // in appPropertiesConfig context parameter of the web module
                "${project.rootPackageDirectory}/web-app.properties",
                // Add this file which is located in CUBA and defines some properties
                // specifically for test environment. You can replace it with your own
                // or add another one in the end.
                "com/haulmont/cuba/web/testsupport/test-web-app.properties"
        );
    }

    public static class Common extends ${project.testContainerPrefix}WebTestContainer {

        // A common singleton instance of the test container which is initialized once for all tests
        public static final ${project.testContainerPrefix}WebTestContainer.Common INSTANCE = new ${project.testContainerPrefix}WebTestContainer.Common();

        private static volatile boolean initialized;

        private Common() {
        }

        @Override
        public void before() throws Throwable {
            if (!initialized) {
                super.before();
                initialized = true;
            }
            setupContext();
        }

        @Override
        public void after() {
            cleanupContext();
            // never stops - do not call super
        }
    }
}

File: src/main/resources/com/haulmont/cuba/cli/cubaplugin/templates/project/7.2/modules/web/test/$[project.rootPackage]/${project.testContainerPrefix}WebTestContainer.kt
package ${project.rootPackage}

import com.haulmont.cuba.web.testsupport.TestContainer
import java.util.*

open class ${project.testContainerPrefix}WebTestContainer : TestContainer() {

    init {
        appComponents = ArrayList(Arrays.asList(
                "com.haulmont.cuba" // add CUBA add-ons and custom app components here
        ))
        appPropertiesFiles = Arrays.asList( // List the files defined in your web.xml
                // in appPropertiesConfig context parameter of the web module
                "${project.rootPackageDirectory}/web-app.properties",  // Add this file which is located in CUBA and defines some properties
                // specifically for test environment. You can replace it with your own
                // or add another one in the end.
                "com/haulmont/cuba/web/testsupport/test-web-app.properties"
        )
    }

    class Common private constructor() : ${project.testContainerPrefix}WebTestContainer() {

        @Throws(Throwable::class)
        override fun before() {
            if (!initialized) {
                super.before()
                initialized = true
            }
            setupContext()
        }

        override fun after() {
            cleanupContext()
            // never stops - do not call super
        }

        companion object {

            // A common singleton instance of the test container which is initialized once for all tests
            val INSTANCE = Common()
            @Volatile
            private var initialized = false
        }

    }
}